lib/AST/RequirementMachine/RewriteSystem.cpp

```diff
@@ -15,6 +15,7 @@
 #include "llvm/Support/raw_ostream.h"
 #include <algorithm>
 #include <vector>
+#include "PropertyMap.h"
```

Comment:
This means that RewriteSystem.cpp and PropertyMap.cpp include each other’s header, which is a bit funny. Perhaps the diagnostics methods could go in a new Diagnostics.cpp, along with the bits from RequirementLowering.cpp and the rule::getPropertyRequirement() method